### PR TITLE
[Staticroute] allow to define non-required parameters

### PR DESCRIPTION
**Pull request by dpfaffenbauer**
As a Bundle provider, I wan't to supply the Developer with default routes and I want to use these routes within my Bundle. In various use-cases, these static routes get changed. (eg. from `/en/shop` to just `/`). Since my routes require certain parameters (eg. `_locale`), which get set on route-generation, Pimcore attaches these Params as GET Parameters `?_locale=en`, which is not necessary in some cases. 

So basically we don't want to attach certain parameters as GET Parameters when not directly set within the reverse route.

This PR allows to define non-required parameters for a static route.

### CODE REVIEW

**Comment by dpfaffenbauer** [comment]
PS: UI is still missing, I wanna know if you're happy with it

**Comment by brusch** [comment]
Basically the idea would be fine for me, but I'm not entirely happy with the naming, because actually the parameters are not "non-required" but more "ignored" on assembling the link, right? 
I'm also not really happy with "ignored" but maybe there are other ideas 😊

**Comment by aarongerig** [comment]
Those parameters would only be ignored, if they are not set within the reverse route. Because of that they are **not always** ignored, but rather not always mandatory (= non-required). Therefore I'd suggest the following naming alternatives:

- `optional` parameters
- `arbitrary` parameters

**Comment by brusch** [comment]
Hmm, they are just ignored for the `query`, so maybe `ignoredQueryParams`?

**Comment by dpfaffenbauer** [comment]
So, what are we using now?

**Comment by brusch** [comment]
I'd go for `ignoredQueryParams` as it is more specific and meaningful in my opinion 😐

**Comment by brusch** [comment]
🏓😊

**Comment by dpfaffenbauer** [comment]
will do next week ;)

**Comment by brusch** [comment]
Closed - stale PR - please feel free to continue the discussion or create a follow-up PR once this topic get's active again. Thanks!
